datafusion/core/src/physical_plan/aggregates/row_hash.rs

```diff
@@ -332,8 +450,14 @@ impl Stream for GroupedHashAggregateStream {
                             self.input_done = true;
                             self.group_ordering.input_done();
                             let timer = elapsed_compute.timer();
-                            let batch = extract_ok!(self.emit(EmitTo::All));
-                            self.exec_state = ExecutionState::ProducingOutput(batch);
+                            if self.spill_state.spills.is_empty() {
+                                let batch = extract_ok!(self.emit(EmitTo::All, false));
+                                self.exec_state = ExecutionState::ProducingOutput(batch);
+                            } else {
+                                // If spill files exist, stream-merge them.
+                                extract_ok!(self.update_merged_stream());
+                                self.exec_state = ExecutionState::ReadingInput;
```

Comment:
Don't we need `self.input_done = false;`? I think for `ExecutionState::ReadingInput`, we assert it should be false.

Reply:
Nvm, saw it is set in `update_merged_stream`.